### PR TITLE
feat: week 1

### PR DESCRIPTION
**Pull request by francescor-tuidi**
Deliverable for this week:

- [x] create a Python package that defines preprocessing steps for your chosen dataset so that it can be used in a machine learning project.

- [x] create a feature branch in your personal repository, and create a PR to main for us to review your code.

Useful links:
- Code used in the lecture: https://github.com/end-to-end-mlops-databricks/module-code-examples
- Data used in the course: https://www.kaggle.com/c/house-prices-advanced-regression-techniques/data

## Release Notes

- **New Features**
	- Introduced a new configuration file for Databricks to manage project assets.
	- Added functionality for loading and processing configuration data in YAML format.
	- Implemented a `DataProcessor` class for handling video game sales data, including methods for loading, preprocessing, and splitting data.
	- Added a `VideoGameModel` class for predicting video game sales with methods for training and evaluation.
	- New decorator for logging execution time of functions.

- **Bug Fixes**
	- Improved pre-commit checks and dependency installation in the CI workflow.

- **Documentation**
	- Cleaned up whitespace in the README file for better formatting.
	- Added initialization files and docstrings in various directories for clarity.

- **Chores**
	- Updated Python version requirements and dependencies in the project configuration files.

### CODE REVIEW

**Review by basakeskili — DISMISSED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -3,37 +3,43 @@ name = "mlops-with-databricks"
...
+    "mlflow>=2.16.0, <3",
+    "numpy>=1.26.4, <2",
+    "pandas>=2.2.2, <3",
+    "pyarrow>=15.0.2, <16",
+    "cffi>=1.17.1, <2",
+    "scipy>=1.14.1, <2",
+    "matplotlib>=3.9.2, <4",
+    "databricks-feature-engineering>=0.6, <1",
+    "pre-commit>=4.0.1",
+    "pytest>=8.3.3",
```
Comment (mvechtomova) [inline pyproject.toml:19]: pytest is not really a project dependency (you do not need it to run code on Databricks), so is pre-commit, those must be in optional dependencies

File: src/utilies/decorators.py
Comment (mvechtomova) [inline src/utilies/decorators.py:1]: nice job on decorators!

**Review by mvechtomova — DISMISSED** [review]
good job! just fix the dependency part in the next submission, I will approve now to unblock you

File: tests/video_games_sales/test_data_processor.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+"""Tests for the DataProcessor class."""
+
+import numpy as np
+
+from src.video_game_sales.data_processor import DataProcessor
```
Comment (mvechtomova) [inline tests/video_games_sales/test_data_processor.py:5]: typically, inside src folder you have another folder with project name  (for example, video_games)

**Review by mvechtomova — APPROVED** [review]
nice job! the only comment is about the project structure, create a folder video_games inside your src folder, and put all the files there - that would be your package folder.

**Pull request merged**